safe/processing/inasafe_processing_algorithm.py

```diff
@@ -0,0 +1,144 @@
+# coding=utf-8
+
+__copyright__ = "Copyright 2019, Kartoza"
+__license__ = "GPL version 3"
+__email__ = "rohmat@kartoza.com"
+__revision__ = "$Format:%H$"
+
+import os
+import tempfile
+
+from PyQt5.QtCore import QCoreApplication, QDate, QSettings
+from PyQt5.QtWidgets import QDateEdit
+from processing.gui.wrappers import WidgetWrapper
+from qgis.core import (
+    QgsProcessing,
+    QgsFeatureSink,
+    QgsProcessingAlgorithm,
+    QgsProcessingParameterFeatureSource,
+    QgsProcessingParameterFeatureSink,
+    QgsProcessingParameterRasterDestination,
+    QgsProcessingParameterEnum,
+    QgsProcessingParameterString,
+    QgsCoordinateReferenceSystem,
+    QgsProcessingParameterNumber)
+
+
+class InaSAFEReportGenerator(QgsProcessingAlgorithm):
+    """Extended QgsProcessingAlgorithm class for generating InaSAFE report.
+
+    All Processing algorithms should extend the QgsProcessingAlgorithm
+    class.
+    """
+    # Constants used to refer to parameters and outputs. They will be
+    # used when calling the algorithm from another algorithm, or when
+    # calling from the QGIS console.
+
+    INPUT = 'INPUT'
+    TEMPLATE = 'TEMPLATE'
+    OUTPUT = 'OUTPUT'
+
+    def tr(self, string):
+        """Translate string on processing context."""
+        return QCoreApplication.translate('Processing', string)
```

Comment:
This would create another context in Transifex, why not using the normal `tr` already in InaSAFE?